### DISCUSSION TITLE
2025 Roadmap

### DISCUSSION THREAD

**Discussion by MarcoGorelli** [narwhals-dev/narwhals#1370] [Announcements]
I'm opening this to give a bit more visibility to the ecosystem about our plans

I have 3 main goals for 2025:
1. **Adoption**: keep looking out for projects which might benefit from Narwhals, and help with Narwhalifying them. So far, each time we've done it, it's highlighted several areas of improvement. I think it's the most valuable kind of contribution. For anyone interested in contributing to Narwhals, trying to Narwhalify a project and then addressing any shortcomings in Narwhals is probably the best way to get started
2. **Laziness**: most (all?) libraries currently using Narwhals are using eager execution. We have the potential to enable so much more. We should aim to have a solid lazy-only layer of support, which DuckDB and PySpark could fall into. Initially, this should exclude all row-order-dependent operations
3. **Standardisation**: I'd like to aim to publish a subset of the Narwhals spec in [JOSS](https://joss.theoj.org/). If we publish that, along with a reference implementation, then it should provide any dataframe library with the tools such that they can become Narwhals-compliant

Keep in mind that this is a community-driven project. Some of us do get some limited amount of time to contribute during work hours - and we're very grateful for this - but there's no Narwhals company or anything like that

**Comment by dangotbanned** [comment]
pinging @binste, in case this sounds interesting to you

It seems you've been quite busy digging into `duckdb` & `pyspark` recently https://github.com/duckdb/duckdb/discussions/14525

**Reply by binste** [reply]
Thank you @dangotbanned, appreciate it :)

**Comment by srivarra** [comment]
A [simple-icons](https://simpleicons.org/) basic svg representing the project would be pretty cool for adoption, that way users can create markdown / html shields.io badges and put it in their own projects to increase visibility. Pandas and polars seem to have some as well.

**Comment by MarcoGorelli** [comment]
2025 has ended

I think we did quite well on points 1 and 2

On point 3, the plugin system is real and ready, but we never published it (and I never looked into JOSS much). More to come in this space anyway!
